testsuites/benchmark/src/main/java/org/eclipse/rdf4j/benchmark/rio/util/DataSetGenerator.java

```diff
@@ -0,0 +1,422 @@
+	private static final String WORD_DIR = "generated-data/words.txt";
+	private static final int TOTAL_PREDS = 500;
+	private static final int AVG_GRAPH_SIZE = 500;
+	private final ValueFactory factory;
+	private final StringBuilder valueBuilder;
+	private final AtomicInteger contextCount = new AtomicInteger(0);
+	private final AtomicInteger maxCurrentContextCount = new AtomicInteger(0);
+	private ArrayList<String> dictionary;
+	private ArrayList<IRI> predicateIRIs;
+	private ArrayList<IRI> contextIRIs;
+	private Random randomResource;
+	private Random randomProbability;
+	private BNode prevBNode;
+	private IRI currentContext;
+	private int totalWords;
+	private int avgWordSize;
+
+	public DataSetGenerator() {
+		this.factory = SimpleValueFactory.getInstance();
+		this.valueBuilder = new StringBuilder();
+		this.randomResource = new Random(64);
+		this.randomProbability = new Random(64);
+		this.avgWordSize = 0;
+		fetchWords();
+		generatePredicateList();
+	}
+
+	public File generateStatementsFile(RDFFormat format, int percentBNodes, int percentLiterals,
+			int textMinLength, int textMaxLength, int totalStatements, boolean isTextOnly,
+			boolean withContext) throws IOException {
+		assert percentBNodes + percentLiterals <= 100 && totalStatements > 0 && textMaxLength > 0
+				&& textMinLength < textMaxLength;
+		File file = createFile(format);
+		if (format.equals(RDFFormat.BINARY)) {
+			FileOutputStream out = new FileOutputStream(file);
+			if (withContext) {
+				generateContextList(totalStatements);
+				writeBinaryStatementsWithContext(out, percentBNodes, percentLiterals, textMinLength, textMaxLength,
+						totalStatements, isTextOnly);
+				maxCurrentContextCount.set(0);
+				contextCount.set(0);
+				currentContext = null;
+			} else {
+				writeBinaryStatementsWithoutContext(out, percentBNodes, percentLiterals, textMinLength,
+						textMaxLength,
+						totalStatements, isTextOnly);
+			}
+		} else {
+			BufferedWriter out = new BufferedWriter(
+					new OutputStreamWriter(new FileOutputStream(file), StandardCharsets.UTF_8));
+			if (withContext) {
+				generateContextList(totalStatements);
+				writeStatementsWithContext(out, format, percentBNodes, percentLiterals, textMinLength, textMaxLength,
+						totalStatements, isTextOnly);
+				maxCurrentContextCount.set(0);
+				contextCount.set(0);
+				currentContext = null;
+			} else {
+				writeStatementsWithoutContext(out, format, percentBNodes, percentLiterals, textMinLength,
+						textMaxLength,
+						totalStatements, isTextOnly);
+			}
+		}
+		prevBNode = null;
+		randomResource = new Random(64);
+		randomProbability = new Random(64);
+		return file;
+	}
+
+	private File createFile(RDFFormat format) throws IOException {
+		File file = new File(
+				Strings.append(File.separator).to(System.getProperty("java.io.tmpdir")) + UUID.randomUUID() + "."
+						+ format.getDefaultFileExtension());
+		assert file.createNewFile();
```

Comment:
This will only create a new file if asserts are enabled. Asserts should probably be left disabled during benchmarking so we don't skew performance.